temporalio/worker/_workflow.py

```diff
@@ -193,6 +206,47 @@ async def _handle_activation(
     ) -> None:
         global LOG_PROTOS
 
+        def get_traceback_for_thread(
+            thread_id: int,
+        ) -> Tuple[Optional[TracebackType], bool]:
+            """Take a thread id and attempt to construct a traceback from its frames.
+
+            Args:
+                thread_id: Thread ID to be checked.
+
+            Returns:
+                A tuple containing:
+                    0: A TracebackType if the supplied thread id is valid, None otherwise.
+                    1: A bool that determines whether the error message should warn about a possibly incomplete traceback.
+            """
+            # retrieve all frames
+            frame = sys._current_frames().get(thread_id)
```

Comment:
Are we sure this is the appropriate API to use? It has an underscore prefix. Should we be using [`inspect.stack()`](https://docs.python.org/3/library/inspect.html#inspect.stack)?

Reply:
While it has an underscore prefix, it is documented at https://docs.python.org/3/library/sys.html#sys._current_frames and it is the only way we could find to get a stack trace of a thread by its ID outside of that thread (there are some C API options, but they are rough too).